src/operations/hulls/hullGeom2.js

```diff
@@ -16,13 +16,18 @@ const hullGeom2 = (...geometries) => {
 
   // extract the unique points from the geometries
   let uniquepoints = []
-  geometries.forEach((geometry) => {
-    let sides = geom2.toSides(geometry)
-    sides.forEach((side) => {
-      let index = uniquepoints.findIndex((unique) => vec2.equals(unique, side[0]))
-      if (index < 0) uniquepoints.push(side[0])
-    })
-  })
+  let found = []
```

Comment:
?

Reply:
This code is actually from V1 hull(), which creates a hash from the strings. See V1 src/api/ops-transformations.js

This is about 10X faster then the nested forEach code of V2.
Figure that!

I'm not sure why there's a huge performance hit but think there may be two issues.
- string values are natively supported, so lookups are FAST
- closure may be occuring in the nested forEach loops, so loops are SLOW

Thoughts?

Reply:
If there are no solutions for the nesting of array functions then javascript kind of fails as a language. :(

Reply:
Ouch ! I am not sure if the nested for each is the reason for the slowdown: I see two more likely culprits:
- the use of findIndex for each item in the nested loop
- possible the vec2.equals aspect

I am fine with hashed but
- string based is meh
- the code readability is heavilly reduced (not your fault)

Reply:
I have some ideas and will try tomorrow.

Reply:
I failed. The best solution is the nested for loops.